src/main/php/xp/scriptlet/Runner.class.php

```diff
@@ -196,9 +204,17 @@ public function run($url= '/') {
         $instance->setTrace($cat);
       }
       $instance->init();
-    
+
+      // Set up request and response
+      $request= $instance->request();
+      $request->method= $_SERVER['REQUEST_METHOD'];
+      $request->env= $_ENV;
```

Comment:
I'd really rather change that all the way to constructor parameters. For that, I'm currently working on getting rid of request and response subclassing in favor of wrapping.

In the future, we'd write:

```php
$request= new ScriptletRequest($method, $url, $headers, $env);
$wrapped= new XmlScriptletRequest($request);
```